internal/driver/controllerserver_helper.go

```diff
@@ -0,0 +1,328 @@
+// to gigabytes.
+// This function should be used when converting a CSI RPC type's capacity range
+// to a value that the Linode API will understand.
+func bytesToGB(numBytes int64) int { return int(numBytes >> 30) }
+
+// gbToBytes is a convenience function that converts gigabytes to bytes.
+// This function is typically going to be used when converting
+// [github.com/linode/linodego.Volume.Size] to a value that works with the CSI
+// RPC types.
+func gbToBytes(gb int) int64 { return int64(gb << 30) }
+
+const (
+	// WaitTimeout is the default timeout duration used for polling the Linode
+	// API, when waiting for a volume to enter an "active" state.
+	WaitTimeout = 5 * time.Minute
+
+	// CloneTimeout is the duration to wait when cloning a volume through the
+	// Linode API.
+	CloneTimeout = 15 * time.Minute
+)
+
+// waitTimeout is a convenience function to get the number of seconds in
+// [WaitTimeout].
+func waitTimeout() int {
+	return int(WaitTimeout.Truncate(time.Second).Seconds())
+}
+
+// cloneTimeout is a convenience function to get the number of seconds in
+// [CloneTimeout].
+func cloneTimeout() int {
+	return int(CloneTimeout.Truncate(time.Second).Seconds())
+}
+
+const (
+	// VolumeTags is the parameter key used for passing a comma-separated list
+	// of tags to the Linode API.
+	VolumeTags = Name + "/volumeTags"
+
+	// PublishInfoVolumeName is used to pass the name of the volume as it exists
+	// in the Linode API (the "label") to [NodeStageVolume] and
+	// [NodePublishVolume].
+	PublishInfoVolumeName = Name + "/volume-name"
+
+	// VolumeTopologyRegion is the parameter key used to indicate the region
+	// the volume exists in.
+	VolumeTopologyRegion string = "topology.linode.com/region"
+
+	// devicePathKey is the key used in the publish context map when a volume is
+	// published/attached to an instance.
+	devicePathKey = "devicePath"
+)
+
+// canAttach indicates whether or not another volume can be attached to the
+// Linode with the given ID.
+//
+// Whether or not another volume can be attached is based on how many instance
+// disks and block storage volumes are currently attached to the instance.
+func (s *ControllerServer) canAttach(ctx context.Context, instance *linodego.Instance) (canAttach bool, err error) {
+	log := logger.GetLogger(ctx)
+	log.V(4).Info("Checking if volume can be attached", "instance_id", instance.ID)
+
+	// Get the maximum number of volume attachments allowed for the instance
+	limit, err := s.maxVolumeAttachments(ctx, instance)
+	if err != nil {
+		return false, err
+	}
+
+	// List the volumes currently attached to the instance
+	volumes, err := s.client.ListInstanceVolumes(ctx, instance.ID, nil)
+	if err != nil {
+		return false, errInternal("list instance volumes: %v", err)
+	}
+
+	// Return true if the number of attached volumes is less than the limit
+	return len(volumes) < limit, nil
+}
+
+// maxVolumeAttachments calculates the maximum number of volumes that can be attached to a Linode instance,
```

Comment:
while we are at it, can you remove the limits and limits_tests modules? I believe this is being used in favor of the function in limits

Reply:
Also can we rename this to `maxAllowedVolumeAttachments`?